### PR TITLE
Fix typo in `#define ST25DV_MBLEN_DYN_REG`

### PR DESCRIPTION
**Pull request by bigjosh**

### CODE REVIEW

**Review by fpistm — COMMENTED** [review]
Hi @bigjosh 
I understand you think it is a typo error as other use `MB_`
Anyway this file comes from ST BSP.
https://github.com/search?q=org%3ASTMicroelectronics+ST25DV_MBLEN_DYN_REG&type=code

If you change the name here you should also change it in the code:
https://github.com/stm32duino/ST25DV/blob/33ad94d8d8c287e215ed7a7965621ef8089e04e1/src/ST25DV/st25dv.c#L1462
https://github.com/stm32duino/ST25DV/blob/33ad94d8d8c287e215ed7a7965621ef8089e04e1/src/ST25DV/st25dv.c#L1480

Moreover it is the name reference in [AN4977: ST25DV applications based on STM32Cube](https://www.st.com/content/ccc/resource/technical/document/application_note/group0/7c/a0/99/1c/21/b1/46/b0/DM00348299/files/DM00348299.pdf/jcr:content/translations/en.DM00348299.pdf) 

image

So I don't think it is a good idea to change it and I don't think you test this change as it should not compile due to the source code not updated.

**Comment by bigjosh** [comment]
The name of the register is `MB_LEN_DYN` in the canonical datasheets....

image

… and all other registers following this naming convention, so I think a typo. I understand not wanting to break existing code so maybe could add an additional `#define` for the incorrect version with a deprecation warning.

**Comment by fpistm** [comment]
Hi @bigjosh 

> maybe could add an additional `#define` for the incorrect version with a deprecation warning.

Agree. @cparata any thought about this ?

**Comment by cparata** [comment]
Hi all,
I agree, but we need to update the PR in order to replace all the references to the old define and then add only a deprecated definition for backward compatibility. @bigjosh , could you update the PR accordingly?
Thanks in advance for your contribution.
Best Regards,
Carlo

File: src/ST25DV/st25dv.h [hunk 1]
```diff
@@ -453,7 +453,7 @@ typedef struct {
 /** @brief ST25DV Mailbox control dynamic register address. */
 #define ST25DV_MB_CTRL_DYN_REG               0x2006
 /** @brief ST25DV Mailbox message length dynamic register address. */
-#define ST25DV_MBLEN_DYN_REG                 0x2007
+#define ST25DV_MB_LEN_DYN_REG                 0x2007
```
Comment (fpistm) [inline src/ST25DV/st25dv.h:456]:
```suggestion
#define ST25DV_MB_LEN_DYN_REG                 0x2007
/** @brief For backward compatibility */
#define ST25DV_MBLEN_DYN_REG                  ST25DV_MB_LEN_DYN_REG
```

**Review by fpistm — CHANGES_REQUESTED** [review]

**Review by fpistm — APPROVED** [review]

**Comment by fpistm** [comment]
@cparata 
I've updated the PR and add a deprecation warning which produce this if old define is used:
```console
C:\STM32\arduino\arduino-1.8.16\portable\sketchbook\libraries\STM32duino_ST25DV\src\ST25DV\st25dv.c: In function 'ST25DV_i2c_ReadMailboxRegister':
C:\STM32\arduino\arduino-1.8.16\portable\sketchbook\libraries\STM32duino_ST25DV\src\ST25DV\st25dv.c:1462:13: warning: 'ST25DV_MBLEN_DYN_REG' definition is deprecated use 'ST25DV_MB_LEN_DYN_REG' instead
 1462 |   if ((TarAddr < ST25DV_GPO_DYN_REG) || (TarAddr > ST25DV_MBLEN_DYN_REG)) {
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~         
```

**Review by cparata — APPROVED** [review]

**Comment by fpistm** [comment]
Thank you @cparata

**Pull request merged**